source/SpinalHDL/Data types/bool.rst

```diff
@@ -9,7 +9,52 @@ Bool
 Description
 ^^^^^^^^^^^
 
-The ``Bool`` type corresponds to a boolean value (True or False).
+The ``Bool`` type corresponds to a boolean value (True or False) or a single bit/wire
+used in a hardware design.  While named similarly it should not be confused with
+Scala `Boolean` type which does not describe hardware but truth values in the Scala
+generator code.
+
+An important concept and rule-of-thumb to understand is that the Scala
+`Boolean` type is used in places where ahead-of-time HDL code-generation
+decision making is occuring in Scala code.  Like any regular program it affects
```

Comment:
"It is a condition during generation of the hardware and does not become hardware."?
The current sentence is far from straightforward

Reply:
I am unable to relate your quoted sentence/question to the original text.

I do not understand your sentence in quotes, and do not think it is a better suggestion.  It lacks context, it doesn't explain what is a condition of what, it repeats the use of the hardware term in a bizarre reference to itself.

Unfortunately none of the concepts being conveyed are straightforward, it is just the nature of the complexity that exists and the elements of underlying knowledge needed to digest new concepts.  But we can try improve clarify.